hierarchy/store_test.go

```diff
@@ -67,6 +88,198 @@ func TestStore_BuildHierarchy(t *testing.T) {
+				return nil
+			},
+			CloneRelationshipsFromIDsFunc: func(ctx context.Context, attempt int, instanceID string, dimensionName string, ids map[string]string) error {
+				return nil
+			},
+			GetHierarchyNodeIDsFunc: func(ctx context.Context, attempt int, instanceID string, dimensionName string) (map[string]string, error) {
+				return testCreatedNodeIds, nil
+			},
+			RemoveCloneEdgesFromSourceIDsFunc: func(ctx context.Context, attempt int, ids map[string]string) error {
+				return nil
+			},
+			SetNumberOfChildrenFromIDsFunc: func(ctx context.Context, attempt int, ids map[string]string) error {
+				return nil
+			},
+		}
+
+		store := &hierarchy.Store{DB: mockDB}
+
+		Convey("When BuildHierarchy is called", func() {
+
+			err := store.BuildHierarchy(testInstanceID, testCodeListID, testDimensionName)
+
+			Convey("Then the HierarchyExists function is called with the expected parameters", func() {
+				calls := mockDB.HierarchyExistsCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].InstanceID, ShouldEqual, testInstanceID)
+				So(calls[0].Dimension, ShouldEqual, testDimensionName)
+			})
+
+			Convey("Then CreateInstanceHierarchyConstraints is called with the expected paramters", func() {
+				calls := mockDB.CreateInstanceHierarchyConstraintsCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].InstanceID, ShouldEqual, testInstanceID)
+				So(calls[0].DimensionName, ShouldEqual, testDimensionName)
+			})
+
+			Convey("Then GetCodesWithData is called with the expected paramters", func() {
+				calls := mockDB.GetCodesWithDataCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].InstanceID, ShouldEqual, testInstanceID)
+				So(calls[0].DimensionName, ShouldEqual, testDimensionName)
+			})
+
+			Convey("Then GetGenericHierarchyNodeIDs is called with the expected paramters", func() {
+				calls := mockDB.GetGenericHierarchyNodeIDsCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].CodeListID, ShouldEqual, testCodeListID)
+				So(calls[0].Codes, ShouldResemble, testCodes)
+			})
+
+			Convey("Then GetGenericHierarchyAncestriesIDs is called with the expected paramters", func() {
+				calls := mockDB.GetGenericHierarchyAncestriesIDsCalls()
+				So(calls, ShouldHaveLength, 1)
+				So(calls[0].CodeListID, ShouldEqual, testCodeListID)
+				So(calls[0].Codes, ShouldResemble, testCodes)
+			})
+
+			Convey("Then CreateHasCodeEdges is called twice, without repeated items even if some nodes are associated to a requested code and an ancestry at the same time", func() {
+				calls := mockDB.CreateHasCodeEdgesCalls()
+				So(calls, ShouldHaveLength, 2)
+				So(calls[0].CodeListID, ShouldEqual, testCodeListID)
+				So(calls[0].CodesById, ShouldResemble, map[string]string{
+					"node1": "code1",
+					"node2": "code2",
+					"node3": "code3",
+				})
+				So(calls[1].CodeListID, ShouldEqual, testCodeListID)
+				So(calls[1].CodesById, ShouldResemble, map[string]string{
+					"node4": "code4",
+					"node5": "code5",
```

Comment:
These expected `CodesById` maps are used multiple times in this test - consider assigning to a var at the start of the test and using it when applicable - keeps the test concise plus is removes duplication and makes the test easier to maintain.